Makefile

```diff
@@ -35,11 +35,23 @@ clean:
 	rm -rf ./bin
 
 ci-preflight-checks:
+	$(MAKE) check-pipelinefiles
```

Comment:
I don't think it makes sense to run this on every PR

We only need to execute this when we detect that pipeline files change (and strictly speaking, only the pipline files that have actually changed, if we can determine that)

Reply:
it will be a lot more cumbersome trying to check which pipeline files have changed as they are spread out across the repo.
It does not take a long time to validate all the pipeline files

Reply:
I think we should do whatever work we need to avoid that cumbersome logic, then - e.g., standardizing on names + locations. We try pretty hard to avoid running CI jobs that test unmodified code, so we should be doing the same here.

Reply:
How many directories would we need to check?